js/try.js

```diff
@@ -94,6 +97,58 @@ const setCache = ({
   }
 }
 
+const doDownload = async (downloadButton) => {
+  try {
+    const {
+      identifier,
+    } = getCache();
+
+    let classes = [];
+    for (let i = 0; i <= 35; i++) { // say there's 35 weeks
+      const url = buildWeekURL(identifier, i);
+      downloadButton.textContent = `Fetching week ${i}`;
+      const timetable = await fetchAndParseTimetable(url);
+      const weekClassArray = Object.values(timetable);
+      classes = classes.concat(weekClassArray)
+    }
+
+    const cal = ics();
+    for (let i = 0; i < classes.length; i++) {
```

Comment:
You could use `for (const class of classes)`.

Reply:
Or even `classes.map((class) => { ...code here })`

Reply:
Or even `classes.forEach((class) => { ...code here })`

Reply:
oooh, fancy js - nice one, thanks!

im still used to java :D